crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -125,6 +125,92 @@ impl Ellipse {
     }
 }
 
+/// A primitive shape formed by the region between two circles, also known as a ring.
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[doc(alias = "Ring")]
+pub struct Annulus {
+    /// The inner circle of the annulus
+    pub inner_circle: Circle,
+    /// The outer circle of the annulus
+    pub outer_circle: Circle,
+}
+impl Primitive2d for Annulus {}
+
+impl Default for Annulus {
+    /// Returns the default [`Annulus`] with radii of `0.5` and `1.0`.
+    fn default() -> Self {
+        Self {
+            inner_circle: Circle { radius: 0.5 },
+            outer_circle: Circle { radius: 1.0 },
+        }
+    }
+}
+
+impl Annulus {
+    /// Create a new [`Annulus`] from an inner and outer circle
+    #[inline(always)]
+    pub const fn new(inner_circle: Circle, outer_circle: Circle) -> Self {
+        Self {
+            inner_circle,
+            outer_circle,
+        }
+    }
```

Comment:
Imo this should still just take floats since that's more ergonomic and the circles are somewhat redundant in a constructor:

```rust
// Compare
let annulus = Annulus::new(Circle::new(1.0), Circle::new(2.0));
let annulus = Annulus::new(1.0, 2.0);
```

I wonder if implementing `Into<Circle>` for `f32` would make any sense... That's out of scope for this PR though.

```suggestion
    /// Create a new [`Annulus`] from the radii of the inner and outer circle
    #[inline(always)]
    pub const fn new(inner_radius: f32, outer_radius: f32) -> Self {
        Self {
            inner_circle: Circle::new(inner_radius),
            outer_circle: Circle::new(outer_radius),
        }
    }
```